### PR TITLE
add: asserting for token balance and allowance

### PR DESCRIPTION
**Pull request by Sokel**

### CODE REVIEW

File: insonmnia/node/market.go [hunk 1]
```diff
@@ -399,8 +400,28 @@ func (m *marketAPI) orderLoop(handler *orderHandler) error {
 		log.G(handler.ctx).Info("found order", zap.Int("count", len(orders)))
 	}
 
+	addr := util.PubKeyToAddr(m.remotes.key.PublicKey).String()
+	balance, err := m.remotes.eth.BalanceOf(addr)
+	if err != nil {
+		log.G(handler.ctx).Info("cannot get balance", zap.Error(err), zap.String("address", addr))
```
Comment (3Hren) [inline insonmnia/node/market.go]: `log.G(handler.ctx).Error` instead?

File: insonmnia/node/market.go [hunk 2]
```diff
@@ -399,8 +400,28 @@ func (m *marketAPI) orderLoop(handler *orderHandler) error {
 		log.G(handler.ctx).Info("found order", zap.Int("count", len(orders)))
 	}
 
+	addr := util.PubKeyToAddr(m.remotes.key.PublicKey).String()
```
Comment (3Hren) [inline insonmnia/node/market.go]: Despite of `String` forwards to `Hex` it seems more correctly to explicitly use `Hex`.

File: insonmnia/node/market.go [hunk 3]
```diff
@@ -399,8 +400,28 @@ func (m *marketAPI) orderLoop(handler *orderHandler) error {
...
+	balance, err := m.remotes.eth.BalanceOf(addr)
+	if err != nil {
+		log.G(handler.ctx).Info("cannot get balance", zap.Error(err), zap.String("address", addr))
+		handler.setError(err)
+		return err
+	}
+	allowance, err := m.remotes.eth.AllowanceOf(addr, tsc.DealsAddress)
+	if err != nil {
+		log.G(handler.ctx).Info("cannot get allowance", zap.Error(err), zap.String("address", addr))
```
Comment (3Hren) [inline insonmnia/node/market.go]: The same as above.

File: insonmnia/node/market.go [hunk 4]
```diff
@@ -399,8 +421,21 @@ func (m *marketAPI) orderLoop(handler *orderHandler) error {
 		log.G(handler.ctx).Info("found order", zap.Int("count", len(orders)))
 	}
 
+	balance, allowance, err := m.loadBalanceAndAllowance()
+	if err != nil {
+		log.G(handler.ctx).Error("cannot get orders", zap.Error(err))
```
Comment (sshaman1101) [inline insonmnia/node/market.go]: But you're getting balance and allowance, not orders. Isn't it?

**Pull request merged**